lib/entities/console_game.rb

```diff
@@ -18,59 +21,80 @@ def initialize(name, difficulty)
   def game_progress
     @current_hint = @difficulty[:difficulty][:hints].to_i
     @current_attempt = 1
-    while (0..@difficulty[:difficulty][:attempts].to_i).cover?(@current_attempt)
-      consol_response = @messages.question { I18n.t(USER_ANSWER[:attempt]) }
-      valid_consol_response = input_handle(consol_response, @current_hint)
-      @game_status = @game.compare(valid_consol_response)
-      @current_hint = @game.current_hint
-      break if @game_status == 'win'
-
-      @messages.answer_for_user(@game_status)
-      @current_attempt += 1
+    range = 1..@difficulty[:difficulty][:attempts].to_i
+    while range.cover?(@current_attempt)
+      @game_status = guess_result
+      case @game_status
+      when MESSAGE_GU[:win] then break
+      when USER_ANSWER[:no_hints] then   @messages.answer_for_user(I18n.t(USER_ANSWER[:no_hints]))
+      when Integer
+        send_to_user = I18n.t(USER_ANSWER[:hint_is], hint: @game_status)
+        @messages.answer_for_user(send_to_user)
+      else
+        @messages.answer_for_user(@game_status)
+        @current_attempt += 1
+      end
+    end
+    @messages.game_over(@game.secret_code, statistics, @game_status)
   end
 
-    @messages.game_over(@game.secret_code, statistics)
-  end
+  private
 
   def statistics
-    attempts_used = @difficulty[:difficulty][:attempts] - @current_attempt
+    attempts_used = @current_attempt - 1
     hints_used = @difficulty[:difficulty][:hints] - @current_hint
-    { "user_name": @name, "game_status": @game_status,
+    { "user_name": @name,
       "difficulty": @difficulty[:name],
-      "attempts_total": @attempts_total,
+      "attempts_total": @difficulty[:difficulty][:attempts],
       "attempts_used": attempts_used,
-      "hints_total": @hints_total,
+      "hints_total": @difficulty[:difficulty][:hints],
       "hints_used": hints_used }
   end
 
-  def console_input
-    input_handle(@messages.question { I18n.t(USER_ANSWER[:attempt]) },
-                  @current_hint)
+  def guess_result
+    valid_input = input_handle
+    #binding.pry
+    if valid_input.class == Integer || valid_input == USER_ANSWER[:no_hints] || valid_input == nil
```

Comment:
move conditions like `valid_input.class == Integer || valid_input == USER_ANSWER[:no_hints] || valid_input == nil`
to separate method

Reply:
Give me please example  of Decompose Conditional. I don' t understand this describing https://refactoring.guru/decompose-conditional  how do it.

Reply:
just move complex condition to the separate method:
```
def guess_result
  ...
  if valid_input?(input)
   ...
  end
end

private

def valid_imput?(input)
  valid_input.class == Integer || valid_input == USER_ANSWER[:no_hints] || valid_input == nil
end
```